### ISSUE TITLE
Segfault when starting client without server

### ISSUE THREAD

**Issue by orishavit** [grpc/grpc#15545]
### What version of gRPC and what language are you using?
 grpc-node 1.12.2 on NodeJS 6.14.2
 
### What operating system (Linux, Windows, …) and version?
Debian Linux 8.10 (I used the `node:6` docker image)
 
### What runtime / compiler are you using (e.g. python version or version of gcc)
Node 6.14.2
 
### What did you do?
I used `node:6` docker image

1. Follow the tutorial at https://grpc.io/docs/quickstart/node.html:
  a. `git clone -b v1.12.x https://github.com/grpc/grpc`
  b. `cd grpc/examples/node/dynamic_codegen`
  c. `npm install`
2. Run ONLY the client: `node greeter_client.js`
 
### What did you expect to see?
Some message indicating a timeout
 
### What did you see instead?
`Segmentation fault` 😢 (about 50% of the times, other times nothing happens and the process hangs)

After running `ulimit -c unlimited`, a core dump is also produced:

```
#0  grpc_closure_sched (error=0x4, c=0x226e6f6974706972) at ../deps/grpc/src/core/lib/iomgr/closure.h:297
#1  timer_cancel (timer=<optimized out>) at ../deps/grpc/src/core/lib/iomgr/timer_custom.cc:71
#2  0x00007fd87f8e47ed in custom_connect_callback (socket=0x3244770, error=0x3246460)
    at ../deps/grpc/src/core/lib/iomgr/tcp_client_custom.cc:90
#3  0x00007fd87f8e4724 in tcp_connect (closure=0x3245000, ep=0x3244ff8, interested_parties=<optimized out>, channel_args=0x3245510,
    resolved_addr=0x7ffd6d620350, deadline=20047) at ../deps/grpc/src/core/lib/iomgr/tcp_client_custom.cc:145
#4  0x00007fd87f87daba in chttp2_connector_connect (con=0x3244f90, args=0x7ffd6d620420, result=0x3245270, notify=0x3245280)
    at ../deps/grpc/src/core/ext/transport/chttp2/client/chttp2_connector.cc:215
#5  0x00007fd87f88a85c in continue_connect_locked (c=0x3245240) at ../deps/grpc/src/core/ext/filters/client_channel/subchannel.cc:386
#6  maybe_start_connecting_locked (c=0x3245240) at ../deps/grpc/src/core/ext/filters/client_channel/subchannel.cc:461
#7  0x00007fd87f88b485 in grpc_subchannel_notify_on_state_change (c=0x3245240, interested_parties=0xdeafbeef, state=0x324439c,
    notify=<optimized out>) at ../deps/grpc/src/core/ext/filters/client_channel/subchannel.cc:510
#8  0x00007fd87f8d5db7 in grpc_core::SubchannelData<grpc_core::(anonymous namespace)::PickFirst::PickFirstSubchannelList, grpc_core::(anonymous namespace)::PickFirst::PickFirstSubchannelData>::StartConnectivityWatchLocked (this=<optimized out>)
    at ../deps/grpc/src/core/ext/filters/client_channel/lb_policy/subchannel_list.h:304
#9  0x00007fd87f8d7784 in grpc_core::(anonymous namespace)::PickFirst::PickFirstSubchannelData::ProcessConnectivityChangeLocked (
    this=<optimized out>, connectivity_state=GRPC_CHANNEL_TRANSIENT_FAILURE, error=0x3245b40)
    at ../deps/grpc/src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc:516
#10 0x00007fd87f8d6bad in grpc_core::SubchannelData<grpc_core::(anonymous namespace)::PickFirst::PickFirstSubchannelList, grpc_core::(anonymous namespace)::PickFirst::PickFirstSubchannelData>::OnConnectivityChangedLocked (arg=0x3244300, error=0x3245b40)
    at ../deps/grpc/src/core/ext/filters/client_channel/lb_policy/subchannel_list.h:427
#11 0x00007fd87f89de52 in grpc_combiner_continue_exec_ctx () at ../deps/grpc/src/core/lib/iomgr/combiner.cc:259
#12 0x00007fd87f85a7b4 in grpc_core::ExecCtx::Flush (this=this@entry=0x7ffd6d6205e0) at ../deps/grpc/src/core/lib/iomgr/exec_ctx.cc:131
#13 0x00007fd87f8e482e in ~ExecCtx (this=0x7ffd6d6205e0, __in_chrg=<optimized out>) at ../deps/grpc/src/core/lib/iomgr/exec_ctx.h:96
#14 custom_connect_callback (socket=<optimized out>, error=0x3245ef0) at ../deps/grpc/src/core/lib/iomgr/tcp_client_custom.cc:100
#15 0x000000000129ef47 in uv__stream_connect (stream=0x3245df0) at ../deps/uv/src/unix/stream.c:1384
#16 uv__stream_io (loop=<optimized out>, w=0x3245e78, events=28) at ../deps/uv/src/unix/stream.c:1301
#17 0x00000000012a4d58 in uv__io_poll (loop=loop@entry=0x1e46480 <default_loop_struct>, timeout=4739)
    at ../deps/uv/src/unix/linux-core.c:400
#18 0x0000000001293826 in uv_run (loop=0x1e46480 <default_loop_struct>, mode=UV_RUN_ONCE) at ../deps/uv/src/unix/core.c:368
#19 0x00000000007dade0 in node::Start(int, char**) ()
#20 0x00007fd881b62b45 in __libc_start_main (main=0x7bd150 <main>, argc=2, argv=0x7ffd6d623c18, init=<optimized out>,
    fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffd6d623c08) at libc-start.c:287
#21 0x00000000007bd25d in _start ()
```
 
### Anything else we should know about your project / environment?
Also appears to happen on Node 4, and on Node 8 only hangs but no segfaults

**Comment by kpayson64** [comment]
I wasn't able to reproduce this bug, but from the stack trace I think I have a good idea what the issue is.

**Comment by kpayson64** [comment]
This will be fixed in the next release by #15556
